speech/google/cloud/speech/_gax.py

```diff
@@ -254,13 +250,14 @@ def sync_recognize(self, sample, language_code=None, max_alternatives=None,
                                  uri=sample.source_uri)
         api = self._gapic_api
         api_response = api.sync_recognize(config=config, audio=audio)
-        if len(api_response.results) == 1:
-            results = api_response.results.pop()
-            alternatives = results.alternatives
-            return [Alternative.from_pb(alternative)
-                    for alternative in alternatives]
-        else:
-            raise ValueError('More than one result or none returned from API.')
+
+        # Sanity check: If we got no results back, raise an error.
+        if len(api_response.results) == 0:
+            raise ValueError('No results returned from the Speecn API.')
+
+        # Iterate over any results that came back.
+        for result in api_response.results:
+            yield Result.from_pb(result)
```

Comment:
It was returning a single, flattened result, which was the major bug this PR fixes. I could return a flat list instead, I suppose.